src/picongpu/include/particles/bremsstrahlung/PhotonEmissionAngle.hpp

```diff
@@ -0,0 +1,264 @@
+     *
+     * @param gamma relativistic factor
+     */
+    float_64 maxTheta(const float_64 gamma) const
+    {
+        AimForDelta aimForDelta(photon::MAX_DELTA, gamma);
+
+        std::pair<float_64, float_64> minimum;
+
+        minimum = boost::math::tools::brent_find_minima(
+            aimForDelta,
+            0.0,
+            M_PI,
+            std::numeric_limits<float_64>::digits);
+
+        return minimum.first;
+    }
+
+    /** computes the polar emission angle theta.
+     *
+     * @param delta uniformly distributed random number within [0, 1]
+     * @param gamma relativistic factor
+     * @param maxTheta maximal theta
+     */
+    float_64 theta(const float_64 delta, const float_64 gamma, const float_64 maxTheta) const
+    {
+        AimForDelta aimForDelta(delta, gamma);
+        const float_64 minTheta = 0.0;
+        std::pair<float_64, float_64> minimum;
+
+        minimum = boost::math::tools::brent_find_minima(
+            aimForDelta,
+            minTheta,
+            maxTheta,
+            std::numeric_limits<float_64>::digits);
+
+        return minimum.first;
+    }
+
+public:
+
+    /** Generate lookup table
+     */
+    void init()
+    {
+        namespace pam = PMacc::algorithms::math;
+
+        // there is a margin of one cell to make the linear interpolation valid for border cells.
+        this->dBufTheta = MyBuf(new PMacc::container::DeviceBuffer<float_X, DIM2>(
+            photon::NUM_SAMPLES_DELTA + 1,
+            photon::NUM_SAMPLES_GAMMA + 1));
+
+        PMacc::container::HostBuffer<float_X, DIM2> hBufTheta(this->dBufTheta->size());
+        hBufTheta.assign(float_X(0.0));
+        PMACC_AUTO(curTheta, hBufTheta.origin());
+
+        const float_64 lnMinGamma = pam::log(photon::MIN_GAMMA);
+        const float_64 lnMaxGamma = pam::log(photon::MAX_GAMMA);
+
+        for(uint32_t gammaIdx = 0; gammaIdx < photon::NUM_SAMPLES_GAMMA; gammaIdx++)
+        {
+            const float_64 lnGamma_norm = static_cast<float_64>(gammaIdx) /
+                                          static_cast<float_64>(photon::NUM_SAMPLES_GAMMA - 1);
+            const float_64 gamma = pam::exp(lnMinGamma + (lnMaxGamma - lnMinGamma) * lnGamma_norm);
+            const float_64 maxTheta = this->maxTheta(gamma);
+
+            for(uint32_t deltaIdx = 0; deltaIdx < photon::NUM_SAMPLES_DELTA; deltaIdx++)
+            {
+                const float_64 delta = photon::MAX_DELTA * static_cast<float_64>(deltaIdx) /
+                                       static_cast<float_64>(photon::NUM_SAMPLES_DELTA - 1);
+
+                *curTheta(deltaIdx, gammaIdx) = static_cast<float_X>(this->theta(delta, gamma, maxTheta));
+            }
+        }
+
+        *this->dBufTheta = hBufTheta;
+    }
+
+    /** Return a functor mapping `delta` to the photon emission polar angle `theta`,
+     * where delta is a uniformly distributed random number between zero and one.
```

Comment:
again - please specify the range of the possible random numbers more precisely (if needed)